db/config.cc

```diff
@@ -876,6 +896,13 @@ const db::extensions& db::config::extensions() const {
     return *_extensions;
 }
 
+std::unordered_map<sstring, db::experimental_features::feature>& db::experimental_features::map() {
+    static const auto inst = new std::unordered_map<sstring, db::experimental_features::feature>{
```

Comment:
Why `new`?

Reply:
To avoid the destruction-order problem: https://isocpp.org/wiki/faq/ctors#construct-on-first-use-v2

Reply:
What destruction-order? If it's a static variable it will only be destructed when the program shuts down, if at all. How can the order of this cause us trouble? And of course, with "new", it will never be destructed at all...

Reply:
It only applies if you have a user in some static destructor, which isn't the case.

Reply:
I don't currently have any static constructors or destructors using this, so technically I don't need the function at all -- I could get away with just a static member.  But this is an established best practice, to avoid future bugs and problems.  We like those, don't we? ;)

Reply:
You can find best practices for anything. Another best practice is not to use naked new, and another is not to leak. Yet another (my favorite) is not to use statics. You can avoid violating all best practices here by returning a temporary rather than a reference to a global. If parsing --experimental starts to dominate execution time, we can consider which best practice to violate in order to make it a static.

btw C++20 will avoid these problems with `constinit`.

Reply:
> You can avoid violating all best practices here by returning a temporary 

OK, done.